configure.ac

```diff
@@ -1526,6 +1534,20 @@ dnl and AM_CONDITIONALs (see below)...
 AM_CONDITIONAL(WITH_FREEIPMI, test "${nut_with_freeipmi}" = "yes")
 dnl AM_CONDITIONAL(WITH_OPENIPMI, test "${nut_with_openipmi}" = "yes")
 
+dnl ----------------------------------------------------------------------
+dnl Check for with-gpio
+
+nut_gpio_lib=""
+
+NUT_ARG_WITH([gpio], [build and install GPIO driver], [auto])
+
+dnl ${nut_with_gpio}: any value except "yes" or "no" is treated as "auto".
```

Comment:
I think somewhere here it should check for libgpiod (potentially other equivalents later). 

Typically the check is done as a new method under `m4` directory (copy and adapt an existing file), setting the `LIBGPIO_CFLAGS`, `LIBGPIO_LDFLAGS`, etc. as appropriate, manually or from `pkg-config` data where available.

The `configure.ac` code calls such method, and if things are found and requested mode was "auto" it becomes "yes", otherwise "no". If the request was "yes" but library was not found, it should cause a fatal `AC_MSG_ERROR`.